WordPress/gutenberg
# Spacing Support: Update padding support to allow configurable sides

## Description

This builds upon https://github.com/WordPress/gutenberg/pull/30606 which updates the BoxControl to allow configurable sides. 

In this PR, the spacing support is updated to accept this new configuration. Its inclusion into the `hooks/style.js` is refactored a little as well to bring it in line with other block supports and smooth the addition of margin support in the next PR of this chain.

## How has this been tested?
Manually.

#### Test Instructions
1. Checkout this PR
2. Create a post, adding a group block to it. Also adding a background color makes viewing the padding changes easier.
3. Select the group block and adjust its padding via the sidebar controls. Ensure the padding BoxControl still works as expected.
4. Change the group block's padding support configuration to only allow certain sides e.g. `"padding": [ "top", "bottom" ]`
    <img width="250" alt="Screen Shot 2021-04-09 at 5 12 23 pm" src="https://user-images.githubusercontent.com/60436221/114142896-d1bd1880-9956-11eb-8832-697d77567775.png">
5. Reload the editor. Re-select the group block and confirm the padding box control now only allows configuration of the sides you selected.
6. Save the post and confirm the correct padding values applied to the group block on the frontend

#### Test Instruction - Global Styles
1. Ensure you have a block based theme active and open the site editor
2. If there is no group block present add one
3. Update your theme.json configuration adding settings for the `core/group` block context opting into padding support but only for top/bottom sides.
4. Confirm there are no spacing controls under Root in the Global Styles sidebar
5. Switch to the "By Block Type" tab in the Global Styles sidebar
6. Expand the Group block and you should see the padding box control allowing top/bottom values to be set
7. Update your theme.json `styles` to add `core/group` block context, specifying padding values.
8. Reload the editor and ensure the box control under the Group block styles reflect your theme.json values.
9. Modify these padding values and ensure the Reset button works. https://github.com/WordPress/gutenberg/pull/30606#discussion_r609504393

## Screenshots 
| All Sides | Custom Sides |
|---|---|
| ![PaddingCustomSidesSupport-NoCustomisation](https://user-images.githubusercontent.com/60436221/113999756-a88c8180-989d-11eb-8602-ea5d119be825.gif) | ![PaddingCustomSidesSupport-CustomSides](https://user-images.githubusercontent.com/60436221/113999770-ad513580-989d-11eb-9e9c-a08673c4635f.gif) |


## Types of changes
New feature. Enhancement.

## Checklist:
